polylaue/ui/regions_navigator/dialog.py

```diff
@@ -0,0 +1,399 @@
+            else:
+                self.image_view.getView().removeItem(roi_item)
+
+    def on_mouse_pressed(
+        self, target: pg.GraphicsScene, event: QGraphicsSceneMouseEvent
+    ):
+        self._pressed = True
+
+        pos = self.image_view.getView().mapSceneToView(event.scenePos())
+
+        initial_size = (1, 1)
+
+        position = (round(pos.x()), round(pos.y()))
+
+        row_count = self.roi_model.rowCount()
+        self.roi_model.beginInsertRows(QModelIndex(), row_count, row_count)
+        id = self.roi_manager.add_roi(position, initial_size)
+        self.roi_model.endInsertRows()
+
+        self.current_roi_id = id
+
+        self.create_roi_item(id, position, initial_size)
+
+        self.sigActiveROIChanged.emit(id)
+
+        return True
+
+    def on_mouse_released(
+        self, target: pg.GraphicsScene, event: QGraphicsSceneMouseEvent
+    ):
+        self._pressed = False
+        self.current_roi_id = None
+
+        return True
+
+    def on_mouse_moved(
+        self, target: pg.GraphicsScene, event: QGraphicsSceneMouseEvent
+    ):
+        if not self._pressed or self.current_roi_id is None:
+            return False
+
+        roi_item = self.roi_items[self.current_roi_id]
+        p0 = roi_item.pos()
+        p1 = self.image_view.getView().mapSceneToView(event.scenePos())
+        size = (round(max(0, p1.x() - p0.x())), round(max(0, p1.y() - p0.y())))
+
+        roi_item.setSize(size)
+
+        return True
+
+    def create_roi_item(self, id, position, size):
+        roi_item = pg.RectROI(position, size, **DEFAULT_ROI_ITEM_ARGS)
+        roi_item.setAcceptedMouseButtons(Qt.MouseButton.LeftButton)
+        roi_item.sigRegionChanged.connect(partial(self.on_roi_update, id))
+        roi_item.sigClicked.connect(partial(self.on_roi_click, id))
+        self.roi_items[id] = roi_item
+        self.image_view.getView().addItem(roi_item)
+
+    def remove_roi_item(self, id) -> bool:
+        if id in self.roi_items:
+            roi_item = self.roi_items[id]
+            self.image_view.getView().removeItem(roi_item)
+            del self.roi_items[id]
+
+            return True
+
+        return False
+
+    def on_roi_update(self, id: str, roi_item: pg.ROI):
+        pos = (round(roi_item.pos()[0]), round(roi_item.pos()[1]))
+        size = (round(roi_item.size()[0]), round(roi_item.size()[1]))
+
+        roi = self.roi_manager.get_roi(id)
+
+        if pos == roi['position'] and size == roi['size']:
+            return
+
+        row = self.roi_manager.id_to_index(id)
+        self.roi_manager.update_roi(id, pos, size)
+        top_left = self.roi_model.createIndex(row, 1)
+        bottom_right = self.roi_model.createIndex(row, 4)
+        self.roi_model.dataChanged.emit(top_left, bottom_right, Qt.EditRole)
+
+        self.sigActiveROIChanged.emit(id)
+        self.sigROIModified.emit(id)
+
+    def on_roi_table_update(self, index: QModelIndex, *args):
```

Comment:
What is the `*args` for?

Reply:
If I remember correctly that signal might have additional parameters that I don't need, so I'm catching them with `*args` just in case

Reply:
Can you try changing this function to not have the `*args`? I think that PySide6 actually inspects these functions and won't pass the argument if the function doesn't accept it (so you don't actually have to accept the argument).

Reply:
done